packages/beta/src/api/templateGroups/templateGroupsApi.ts

```diff
@@ -0,0 +1,96 @@
+// Copyright 2020 Cognite AS
+
+import {
+  BaseResourceAPI,
+  CursorAndAsyncIterator,
+  ExternalId,
+} from '@cognite/sdk-core';
+import {
+  ExternalTemplateGroup,
+  TemplateGroup,
+  TemplateGroupFilterQuery,
+} from '../../types';
+
+export class TemplateGroupsApi extends BaseResourceAPI<TemplateGroup> {
+  /**
+   * [Create Template Groups](https://pr-1202.specs.preview.cogniteapp.com/v1.json.html#operation/postApiV1ProjectsProjectTemplategroups)
+   *
+   * Create new Template Groups. The user creating a Template Group is automatically added as an owner of that Template Group,
+   * however more owners can optionally be added by setting the 'owners' field.
+   *
+   * ```js
+   * const templateGroups = [
+   *   { externalId: 'Wells', description: 'Models a Well system', owners: ['user.name@example.com'] },
+   * ];
+   * const createdTemplateGroups = await client.templateGroups.create(templateGroups);
+   * ```
+   */
+  public create = (
+    items: ExternalTemplateGroup[]
+  ): Promise<TemplateGroup[]> => {
+    return super.createEndpoint(items);
+  };
+
+  /**
+   * [Upsert Template Groups](https://pr-1202.specs.preview.cogniteapp.com/v1.json.html#operation/postApiV1ProjectsProjectTemplategroupsUpsert)
+   *
+   * Create or update new Template Groups. The user creating a Template Group is automatically added as an owner of that Template Group,
+   * however more owners can optionally be added by setting the 'owners' field.
+   *
+   * ```js
+   * const templateGroups = [
+   *   { externalId: 'Wells', description: 'Models a Well system', owners: ['user.name@example.com'] },
+   * ];
+   * const upsertedTemplateGroups = await client.templateGroups.upsert(templateGroups);
+   * ```
+   */
+  public upsert = (
+    items: ExternalTemplateGroup[]
+  ): Promise<TemplateGroup[]> => {
+    return super.createEndpoint(items, this.url('upsert'));
+  };
+
+  /**
+   * [Retrieve Template Groups](https://pr-1202.specs.preview.cogniteapp.com/v1.json.html#operation/postApiV1ProjectsProjectTemplategroupsByids)
+   * <!-- or [similar](https://doc.cognitedata.com/api/v1/#operation/getEventByInternalId) -->
+   *
+   * Retrieve Template Groups by externalIds.
+   *
+   * ```js
+   * const templateGroups = await client.templateGroups.retrieve([{externalId: 'abc'}]);
+   * ```
+   */
+  public retrieve = (ids: ExternalId[], ignoreUnknownIds: Boolean = false) => {
```

Comment:
```suggestion
  public retrieve = (ids: ExternalId[], ignoreUnknownIds: boolean = false) => {
```
and better to wrap it in the object like it's done in other apis

Reply:
I see thanks for the suggestion